src/app.rs

```diff
@@ -82,18 +85,45 @@ pub fn start(mut conf: Config) {
                         }
                     }
                 }
-                debug!("{}={}", clone.key, result);
-                output_folder.push(clone.key);
-                match OpenOptions::new().write(true).append(true).create(true).open(&output_folder)
-                    .and_then(|mut file| {
+
+            debug!("{}={}", clone.key, result);
+            output_folder.push(&clone.key);
+            match OpenOptions::new().write(true).append(true).create(true).open(&output_folder)
+                .and_then(|mut file| {
+                    if clone.mappers.is_empty() {
                         file.write(&format!("{} {}", cur_time, &result).as_bytes()[..])
-                    })
-                    {
-                        Ok(_) => (),
-                        Err(e) => {
-                            error!("Error creating file {}, {}", output_folder.display(), e)
+                    } else {
+                        for (i, regex) in clone.mappers
+                            .iter()
+                            .map(|mapper| match mapper {
+                                Mapper::Regex { regex } => regex,
+                            })
+                            .map(|s| Regex::new(s).expect("Parsing regex failed"))
+                            .enumerate()
+                        {
+                            for (j, mtch) in regex
+                                .find_iter(&result)
+                                .enumerate()
+                            {
+                                let out = format!("{time} {mapper}.{nmatch} {text}",
```

Comment:
`1547409378 0.1 0.78` isn't something I'd consider as readable - which of course doesn't matter when the output is parsed by another program.
But this way, we would also have a different metric on every line, which makes analyzing the file more complex, e.g.:
```
1547409378 0.1 8192
1547409378 0.2 1202
1547409378 0.3 6990
```
a program which wants to analyze the logs would need to split them by mapper and match. There's also the problem with optional capture groups, which might mix up the indices of the matches.
Again, named captures should work way better in such a case; the regex is given by the user, so we either enforce the use of named captures, or at least use the name when available.